source/class/qx/ui/list/List.js

```diff
@@ -59,911 +59,980 @@
-
-      this._onModelChange();
-    },
-
-
-    // apply method
-    _applyRowHeight : function(value, old) {
-      this.getPane().getRowConfig().setDefaultItemSize(value);
-    },
-
-    // apply method
-    _applyGroupRowHeight : function(value, old) {
-      this.__updateGroupRowHeight();
-    },
-
-    // apply method
-    _applyLabelPath : function(value, old) {
-      this._provider.setLabelPath(value);
-    },
-
-
-    // apply method
-    _applyIconPath : function(value, old) {
-      this._provider.setIconPath(value);
-    },
-
-
-    // apply method
-    _applyGroupLabelPath : function(value, old) {
-      this._provider.setGroupLabelPath(value);
-    },
-
-
-    // apply method
-    _applyLabelOptions : function(value, old) {
-      this._provider.setLabelOptions(value);
-    },
-
-
-    // apply method
-    _applyIconOptions : function(value, old) {
-      this._provider.setIconOptions(value);
-    },
-
-
-    // apply method
-    _applyGroupLabelOptions : function(value, old) {
-      this._provider.setGroupLabelOptions(value);
-    },
-
-
-    // apply method
-    _applyDelegate : function(value, old) {
-      this._provider.setDelegate(value);
-      this.__buildUpLookupTable();
-    },
-
-
-    // property apply
-    _applyVariableItemHeight : function(value, old) {
-      if(value) {
-        this._setRowItemSize();
-      }
-      else {
-        this.getPane().getRowConfig().resetItemSizes();
-        this.getPane().fullUpdate();
-      }
-    },
-
-
-    /*
-    ---------------------------------------------------------------------------
-      EVENT HANDLERS
-    ---------------------------------------------------------------------------
-    */
-
-
-    /**
-     * Event handler for the resize event.
-     *
-     * @param e {qx.event.type.Data} resize event.
-     */
-    _onResize : function(e) {
-      this.getPane().getColumnConfig().setItemSize(0, e.getData().width);
-    },
-
-
-    /**
-     * Event handler for the model change event.
-     *
-     * @param e {qx.event.type.Data} model change event.
-     */
-    _onModelChange : function(e) {
-      // we have to remove the bindings before we rebuild the lookup table
-      // otherwise bindings might be dispatched to wrong items
-      // see: https://github.com/qooxdoo/qooxdoo/issues/196
-      this._provider.removeBindings();
-      this.__buildUpLookupTable();
-      this._applyDefaultSelection();
-
-      if (e instanceof qx.event.type.Data) {
-        this.fireDataEvent("changeModelLength", e.getData(), e.getOldData());
-      }
-    },
-
-
-    /**
-     * Event handler for the updated event of the 
-     * qx.ui.virtual.layer.WidgetCell layer.
-     *
-     * Recalculates the item sizes in a deffered call,
-     * which only happens if we have variable item heights
-     */
-    _onLayerUpdated: function () {
-      if (this.isVariableItemHeight() === false) {
-        return;
+qx.Class.define("qx.ui.list.List", {
+	extend: qx.ui.virtual.core.Scroller,
```

Comment:
Without reference to coding style, per se, can we avoid use of tabs for indentation? Qooxdoo has traditionally used 2 spaces of indentation. Tabs are "defined" as moving to multiples of 8 spaces, and in fact, that's how it appears, for example, in this github PR. That much indentation makes JavaScript lines really, really long. Do you think you can replace tabs with 2 spaces each in here?